### PR TITLE
UCT/IB/MLX5: No byte_count in KSM entry

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**
## What
Don't set byte_count in KSM

## Why ?
This is reserved field

### CODE REVIEW

File: src/uct/ib/mlx5/dv/ib_mlx5dv_md.c [hunk 1]
```diff
@@ -288,11 +288,6 @@ static ucs_status_t uct_ib_mlx5_devx_reg_atomic_key(uct_ib_md_t *ibmd,
 
     klm = UCT_IB_MLX5DV_ADDR_OF(create_mkey_in, in, klm_pas_mtt);
     for (i = 0; i < list_size; i++) {
-        if (i == list_size - 1) {
-            UCT_IB_MLX5DV_SET(klm, klm, byte_count, length % reg_length);
-        } else {
-            UCT_IB_MLX5DV_SET(klm, klm, byte_count, reg_length);
-        }
         UCT_IB_MLX5DV_SET(klm, klm, mkey, mr->super.ib->lkey);
```
[the remark this answers is not in the document]
Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:291]: byte_count 
Reserved when access mode==Fixed_Buffer_Size, and byte
count is determined by log_entity_size in MKEY Context.

**Review by yosefe — APPROVED** [review]

**Review by evgeny-leksikov — COMMENTED** [review]
maybe add a unit test for the  corner case?

**Comment by evgeny-leksikov** [comment]
> > maybe add a unit test for the corner case?
> 
> Like registering a specific length? @Artemy-Mellanox WDYT?

yep, AFAIU 2GB, for example

**Comment by Artemy-Mellanox** [comment]
> > > maybe add a unit test for the corner case?
> > 
> > 
> > Like registering a specific length? @Artemy-Mellanox WDYT?
> 
> yep, AFAIU 2GB, for example

we can but it won't catch anything because it wasn't really a bug
IMHO we shouldn't add tests which can't fail
